chains/celo/listener_test.go

```diff
@@ -30,3 +90,73 @@ func TestListener_start_stop(t *testing.T) {
 	// Initiate shutdown
 	l.close()
 }
+
+// Testing transaction Block hash
+func TestListener_TransactionBlockHash(t *testing.T) {
+
+	l := createTestListener(t)
+	err := l.start()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create and submit a new transaction and return the signed transaction hash
+	hash := newTransaction(t, l)
+
+	// Get the transaction
+	tx, _, err := l.conn.Client().TransactionByHash(context.Background(), hash)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	receipt, err := waitForTx(l.conn.Client(), tx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Get the block hash of the receipt
+	blockHash, err := l.getTransactionBlockHash(hash)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	retrievedBlock, err := l.conn.Client().BlockByHash(context.Background(), blockHash)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Confirm that the receipt blockhash and the block's blockhash are the same
+	if retrievedBlock.Hash() != receipt.BlockHash {
+		t.Fatalf("block hash are not equal, expected: %x, %x", retrievedBlock.Hash(), receipt.TxHash)
+	}
```

Comment:
I think you just need to compare `blockHash` and `receipt.Blockhash`, there is no need to retrieve the block.